src/giftCards/GiftCardUpdatePage/GiftCardUpdateDetailsCard/GiftCardUpdateInfoCard.tsx

```diff
@@ -0,0 +1,143 @@
+  const navigate = useNavigator();
+  const intl = useIntl();
+  const localizeDate = useDateLocalize();
+
+  const {
+    giftCard: {
+      created,
+      createdByEmail,
+      createdBy,
+      usedByEmail,
+      usedBy,
+      app,
+      product,
+      events
+    }
+  } = useContext(GiftCardDetailsContext);
+
+  const cardIssuedEvent = events.find(
+    ({ type }) => type === GiftCardEventsEnum.ISSUED
+  );
+
+  const getBuyerFieldData = (): {
+    label: MessageDescriptor;
+    name: string;
+    url?: string;
+  } => {
+    if (createdByEmail) {
+      return {
+        label: messages.boughtByLabel,
+        name: createdByEmail
+      };
+    }
+
+    if (app) {
+      return {
+        label: messages.issuedByAppLabel,
+        name: app.name,
+        url: appUrl(app.id)
+      };
+    }
+
+    // createdBy can be either customer or staff hence
+    // we check for issued event
+    if (!!cardIssuedEvent) {
+      return {
+        label: messages.issuedByLabel,
+        name: getFullName(createdBy),
+        url: staffMemberDetailsUrl(createdBy.id)
+      };
+    }
+
+    return {
+      label: messages.boughtByLabel,
+      name: getFullName(createdBy),
+      url: customerUrl(createdBy.id)
+    };
+  };
+
+  const orderData =
+    cardIssuedEvent && cardIssuedEvent.orderId
+      ? getOrderNumberLinkObject({
+          id: cardIssuedEvent.orderId,
+          number: cardIssuedEvent.orderNumber
+        })
+      : null;
+
+  const {
+    label: buyerLabelMessage,
+    name: buyerName,
+    url: buyerUrl
+  } = getBuyerFieldData();
+
+  return (
+    <Card>
+      <CardTitle title={intl.formatMessage(messages.title)} />
+      <CardContent>
+        <Label text={intl.formatMessage(messages.creationLabel)} />
+        <Typography>{localizeDate(created, "DD MMMM YYYY")}</Typography>
+        <CardSpacer />
+
+        <Label text={intl.formatMessage(messages.orderNumberLabel)} />
+        {orderData ? (
+          <Link onClick={() => navigate(orderData.link)}>{orderData.text}</Link>
+        ) : (
+          <Typography>{PLACEHOLDER}</Typography>
+        )}
+        <CardSpacer />
+
+        <Label text={intl.formatMessage(messages.productLabel)} />
+        {product ? (
+          <Link onClick={() => navigate(productUrl(product?.id))}>
+            {product?.name}
+          </Link>
+        ) : (
+          <Typography>{PLACEHOLDER}</Typography>
+        )}
+        <CardSpacer />
+
+        <Label text={intl.formatMessage(buyerLabelMessage)} />
+        {buyerUrl ? (
+          <Link onClick={() => navigate(buyerUrl)}>{buyerName}</Link>
+        ) : (
+          <Typography>{buyerName}</Typography>
+        )}
+        <CardSpacer />
+
+        <Label text={intl.formatMessage(messages.usedByLabel)} />
+        {usedBy ? (
+          <Link onClick={() => navigate(customerUrl(usedBy.id))}>
+            {getFullName(usedBy)}
+          </Link>
+        ) : (
+          <Typography>{usedByEmail || PLACEHOLDER}</Typography>
```

Comment:
Extend `getStringOrPlaceholder` function to take a second `placeholder` argument and use it here